apps/web/src/pages/Applications/ApplicationLayout.tsx

```diff
@@ -112,6 +116,12 @@ const ApplicationPageWrapper = ({ application }: ApplicationPageProps) => {
     }
   }, [currentPage, navigate, nextStepUrl]);
 
+  React.useEffect(() => {
+    if (publishingGroup === PublishingGroup.Iap) {
+      setTheme("iap", mode);
+    }
+  }, [setTheme, publishingGroup, mode]);
```

Comment:
When you leave the application process the theme doesn't switch back.  :smile:
![image](https://user-images.githubusercontent.com/8978655/236829067-be0a0c28-ce60-4352-a9f0-ee032f2612e5.png)

Reply:
Oops, looks like client-side routes don't actually re-render the layout. Made this somewhat hacky hook to fix that in [fix theme not changing on client side route changes](https://github.com/GCTC-NTGC/gc-digital-talent/pull/6521/commits/d5de72cb11b1c34221cf7577a759411596a73d70)

Reply:
Oh, it's almost perfect.  I'm getting this now, though.  
![image](https://user-images.githubusercontent.com/8978655/236877679-20eddc68-a6a8-48d4-9b86-64bb4aefd85c.png)

Reply:
Hmm, the theme seems to dropping out on navigation now.

https://user-images.githubusercontent.com/8978655/236880988-ef83f0d0-c549-44c0-8a44-4b7deb401262.mp4

Reply:
So, I updated the theme changing to be more stable in [stabilize theme changing](https://github.com/GCTC-NTGC/gc-digital-talent/pull/6521/commits/a14cb8ea5a4986681477cd16bc66141256b84074)

However, it flashes very quickly while it sorts out the themes and settles on what it should be showing. This is happening because:

1. Layout loads
2. Sets theme to default
3. Application page loads
4. Retrieves application
5. Checks if it is IAP
6. Sets theme to IAP

The flash is happening in between steps 2-5. I'm trying to think of a way to make this more stable without having to set the theme on _every_ page instead of the layout itself 🤔 Which, as of right now, is the only way I can think of to prevent the flash.

---

Though, taking another look at our `ThemeProvider`, I think this could also use some work. Looks a little wonky and I'm not sure why I wrote it specifically like this. We could leverage our `useLocalStorage` hook and sync the DOM with that instead of all the craziness in this.

Reply:
Oof, I'm in a loop now.  :cry: 
![image](https://github.com/GCTC-NTGC/gc-digital-talent/assets/8978655/86aadd6e-08f6-45b0-905f-7c4534d060cd)

Reply:
Seems good now!